### PR TITLE
Rename CudnnScratchAllocator to DnnScratchAllocator

### PR DESCRIPTION
**Pull request by whchung**
Rename CudnnScratchAllocator as the logic is applicable for not only Cudnn, but
also other DNN algorithm libraries such as MIOpen.

### CODE REVIEW

**Comment by whchung** [comment]
@timshen1 May I ask you to take a look at this PR? It's a precursor to upstream ROCm-specific logic for TensorFlow operators.

Failed CI targets are:
- MacOS Contrib
- MacOS Python2 and CC
- Windows Bazel
- Windows Bazel GPU

From the logs they seem to have nothing to do with this PR, but a configuration issue in CI test infrastructure. I'm wondering if it's possible to manually apply `ready-to-pull` label, after your review and approval?

File: tensorflow/contrib/fused_conv/kernels/fused_conv2d_bias_activation_op.cc [hunk 1]
```diff
@@ -525,7 +525,7 @@ void LaunchFusedConv2DBiasActivationOp<GPUDevice, T, BiasType, ScaleType>::
   static int64 ConvolveScratchSize = GetDnnWorkspaceLimit(
       // default value is in bytes despite the name of the environment variable
       "TF_CUDNN_WORKSPACE_LIMIT_IN_MB", 1LL << 32  // 4GB
-  );
+      );
```
Comment (timshen91) [inline tensorflow/contrib/fused_conv/kernels/fused_conv2d_bias_activation_op.cc]: The indentation looks different from my clang-format output. Which clang-format version do you have? I have 8.0.

Comment (whchung) [reply tensorflow/contrib/fused_conv/kernels/fused_conv2d_bias_activation_op.cc]: It's also different from mine. I just followed what's shown in `Experimental clang-format Check` test to edit the indentation.

Comment (timshen91) [reply tensorflow/contrib/fused_conv/kernels/fused_conv2d_bias_activation_op.cc]: Maybe something is mis-configured. Oh well.

**Review by timshen91 — APPROVED** [review]

**Pull request merged**